pkg/splunk/enterprise/configuration.go

```diff
@@ -778,76 +783,125 @@ func AreRemoteVolumeKeysChanged(client splcommon.ControllerClient, cr splcommon.
+		_, err := checkIfVolumeExists(appFramework.VolList, appFramework.Defaults.VolName)
+		if err != nil {
+			return fmt.Errorf("Invalid Volume Name for Defaults. Error: %s", err)
+		}
 	}
 
 	return nil
 }
 
-// ValidateSplunkSmartstoreSpec checks and validates the smartstore config
-func ValidateSplunkSmartstoreSpec(smartstore *enterprisev1.SmartStoreSpec) error {
-	var err error
+//  isAppFrameworkConfigured checks and returns true if App Framework is configured
+//  App Repo config without any App sources will not cause any App Framework activity
+func isAppFrameworkConfigured(appFramework *enterprisev1.AppFrameworkSpec) bool {
+	return !(appFramework == nil || appFramework.AppSources == nil)
+}
 
-	// Smartstore is an optional config (at least) for now
-	if !isSmartstoreConfigured(smartstore) {
+// ValidateAppFrameworkSpec checks and validates the Apps Frame Work config
+func ValidateAppFrameworkSpec(appFramework *enterprisev1.AppFrameworkSpec, localScope bool) error {
+	var err error
+	if !isAppFrameworkConfigured(appFramework) {
 		return nil
 	}
 
-	numVolumes := len(smartstore.VolList)
-	numIndexes := len(smartstore.IndexList)
-	if numIndexes > 0 && numVolumes == 0 {
-		return fmt.Errorf("Volume configuration is missing. Num. of indexes = %d. Num. of Volumes = %d", numIndexes, numVolumes)
+	scopedLog := log.WithName("ValidateAppFrameworkSpec").WithValues(
+		"localScope: %t", localScope)
+
+	// Todo: sgontla: Enable later. For now, allowing multiple volumes and app sources for UT purpose
+	if len(appFramework.VolList) > 1 || len(appFramework.AppSources) > 2 {
+		//return fmt.Errorf("Invalid App Framework config. Only one Volume, and a max. of two App sources supported")
+	}
+
+	scopedLog.Info("Validating app framework spec")
+
+	//ToDo: sgontla: Do we need to return error, if the poll interval is too small? TBD
+	if appFramework.AppsRepoPollInterval < minAppsRepoPollInterval {
+		//return fmt.Errorf("AppsRepoPollInterval configured value %d, is less than the minimum value of %d seconds", appFramework.AppsRepoPollInterval, minAppsRepoPollInterval)
+		appFramework.AppsRepoPollInterval = minAppsRepoPollInterval
+	} else if appFramework.AppsRepoPollInterval > maxAppsRepoPollInterval {
+		//return fmt.Errorf("AppsRepoPollInterval configured value %d, is more than the max value of %d seconds", appFramework.AppsRepoPollInterval, maxAppsRepoPollInterval)
+		appFramework.AppsRepoPollInterval = maxAppsRepoPollInterval
+	}
+
+	err = validateRemoteVolumeSpec(appFramework.VolList)
+	if err != nil {
+		return err
+	}
+
+	err = validateSplunkAppSources(appFramework, localScope)
+	if err != nil {
+		return err
+	}
+
+	if appFramework.Defaults.VolName != "" {
+		_, err = checkIfVolumeExists(appFramework.VolList, appFramework.Defaults.VolName)
+		if err != nil {
+			return fmt.Errorf("Invalid Volume Name for Defaults. Error: %s", err)
+		}
 	}
 
+	return err
+}
+
+func validateRemoteVolumeSpec(volList []enterprisev1.VolumeSpec) error {
```

Comment:
`func ValidateAppFrameworkSpec()` just validates the config vs. the spec, right?  Not the validity of the configured S3 parameters (i.e. if the S3 endpoint or bucket actually exist, or the creds are correct, etc)

Reply:
That is correct. We are doing that when we are actually using it, which is kind of bad idea, but, just fitting the existing flow, as we need the cr and the controller client to pull any K8 resources, which we are not doing in this flow. Example:-

`func validateLicenseMasterSpec(spec *enterprisev1.LicenseMasterSpec) error {

	err := ValidateAppFrameworkSpec(&spec.AppFrameworkConfig, true)
	if err != nil {
		return err
	}

	return validateCommonSplunkSpec(&spec.CommonSplunkSpec)`

	All we have is application spec. Its not so difficult to pass on the K8 info either :-)